### ISSUE TITLE
Contributing to proj-datumgrid ?

### ISSUE THREAD

**Issue by rouault** [NaturalGIS/ntv2_transformations_grids_and_sample_data#3]
Just discovering this project mentioned in https://lists.osgeo.org/pipermail/proj/2019-February/008217.html

Would you consider contributing those grid collection to the https://github.com/OSGeo/proj-datumgrid/ project that is affiliated with PROJ.4 and serves as a base for providing packages with PROJ compatible grids ? One requirement is that `New grids are accepted into the package as long as they are released under a license that is compatible with the Open Source Definition and the source of the grid is clearly stated and verifiable`

With PROJ 6, those grids could also be automatically be used if inserting (source_crs, target_crs) entries referencing them in the proj.db (if they are not already in the EPSG database)

**Comment by gioman** [comment]
@rouault Hi Even, good afternoon,

> Would you consider contributing those grid collection to the https://github.com/OSGeo/proj-datumgrid/ project that is affiliated with PROJ.4 and serves as a base for providing packages with PROJ compatible grids ? One requirement is that `New grids are accepted into the package as long as they are released under a license that is compatible with the Open Source Definition and the source of the grid is clearly stated and verifiable`

I have no problem in contributing them. About the licensing:

the plugin was first done to support this kind of transformation for Portuguese CRSs... then people noticed the plugin and started ask to add transformations for other countries. I always explicitly told that the grids needed to have a clear (open) license, but in some case was not possible to obtain a clear answer (usually because the publisher didn't care to attach a license to the grids and because it was impossible to reach someone that could possibly give a clear answer and/or take a decision).

Examples: 

very detailed infos but no mention of a license: https://github.com/NaturalGIS/naturalgis_ntv2_transformations_grids_and_sample_data/blob/master/it_rer/NTv2_Pacchetto_base_per_trasformazioni_RER%20v.1.0.1/Trasformazione_Sistemi_Coordinate_v1.0.1.pdf


simple reference to a specific license but no docs: https://github.com/NaturalGIS/naturalgis_ntv2_transformations_grids_and_sample_data/blob/master/it_globo/creative_commons.url


no docs, no license: https://github.com/NaturalGIS/naturalgis_ntv2_transformations_grids_and_sample_data/tree/master/nl

**Comment by rouault** [comment]
Yes, licensing is a pain point. In proj-datumgrid we try to be strict about that, so some grids can't be accepted.
Anyway my point was to make sure you were aware of this effort.
With PROJ 6, there is now a PROJ database where custom transformations can be added referencing grids: e.g. https://github.com/OSGeo/proj.4/pull/1280/commits/2620701a4bb51a20d49e869acddce2167ba791d0
Or if the grids are already referenced in the EPSG database, add the alternate name used and a link to the proj-datumgrid-XXXX package where they can be found : https://github.com/OSGeo/proj.4/pull/1280/commits/6d6ee7fa7d35bf57d1781136102f971cb0ac33d2

**Comment by gioman** [comment]
> Yes, licensing is a pain point. In proj-datumgrid we try to be strict about that, so some grids can't be accepted.

as a local group we are going to ask to the entity that publishes the Portugues grids if they can attach/specify a clear license. But I'm really not sure if they will ever release anything other than the binary grids... 


> Anyway my point was to make sure you were aware of this effort.

Thanks! I will need to explore this important changes/work and see how they can fit the qgis plugin (that I have still to migrate to QGIS3).

**Comment by PedroVenancio** [comment]
Hi @rouault 

As mentioned by @gioman, we asked the entity that publishes the Portugues grids (Direção-Geral do Território), and they had put a CC license in those grids:

https://www.dgterritorio.pt/cartografia_e_geodesia/geodesia/transformacao_de_coordenadas/grelhas_em_ntv2/

So, they can be used in https://github.com/OSGeo/proj-datumgrid/ project. Can you merge them?

Thank you very much!

**Comment by rouault** [comment]
This is great news.

> Can you merge them?

I can do many things, but I don't "scale" very well. So it would be better if you could issue a pull request.
Add the grids in the europe/ subdirectory, update europe/README.EUROPE with the needed information to describe what those grids are for, and add the new grid names in travis/expected_europe.lst (sorted alphabetically)

And so they are properly used by PROJ 6, a pull request to https://github.com/OSGeo/proj.4 updating the https://github.com/OSGeo/proj.4/blob/master/data/sql/grid_alternatives.sql file would be needed. As I see that the grid names in https://www.dgterritorio.pt/cartografia_e_geodesia/geodesia/transformacao_de_coordenadas/grelhas_em_ntv2/ are to the one referenced in the EPSG dataset (https://github.com/OSGeo/proj.4/blob/master/data/sql/grid_transformation.sql#L193), just adding a new entry similar to the one for OSTN15_NTv2_OSGBtoETRS.gsb ( https://github.com/OSGeo/proj.4/blob/master/data/sql/grid_alternatives.sql#L136  ) would be needed

**Comment by rouault** [comment]
I've added more detailed notes in https://github.com/OSGeo/proj-datumgrid/pull/49/files on how to add a new grid
